### PR TITLE
client: add follower option

### PR DESCRIPTION
**Pull request by CabinfeverB**
### What problem does this PR solve?

Issue Number: ref #7431

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Code changes

- Has the configuration change
- Has HTTP API interfaces changed (Don't forget to [add the declarative for the new API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tiup`](https://github.com/pingcap/tiup):
- Need to cherry-pick to the release branch

### Release note

```release-note
None.
```

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -211,6 +212,11 @@ func WithBuckets() GetRegionOption {
 	return func(op *GetRegionOp) { op.needBuckets = true }
 }
 
+// AllowFollowerHandle means that client can send request to follower and let it handle this request.
+func AllowFollowerHandle() GetRegionOption {
```
Comment (rleungx) [inline client/client.go:216]: How about keeping consistent with the HTTP interface and changing it to WithAllowFollowerHandle?

Comment (CabinfeverB) [reply client/client.go:216]: gRPC uses lower case, but HTTP uses MIME format. So I think no need to keep consistent.

**Review by rleungx — APPROVED** [review]

**Review by lhy1024 — APPROVED** [review]

File: client/option.go [hunk 2]
```diff
@@ -88,6 +92,19 @@ func (o *option) setMaxTSOBatchWaitInterval(interval time.Duration) error {
...
 
+// setEnableFollowerHandle set the Follower Handle option.
+func (o *option) setEnableFollowerHandle(enable bool) {
+	old := o.getEnableFollowerHandle()
+	if enable != old {
+		o.dynamicOptions[EnableFollowerHandle].Store(enable)
+	}
+}
+
+// getMaxTSOBatchWaitIntervalgets the Follower Handle enable option.
```
Comment (HuSharp) [inline client/option.go:103]:
```suggestion
// getMaxTSOBatchWaitInterval gets the Follower Handle enable option.
```

**Review by HuSharp — APPROVED** [review]

**Comment by lidaohang** [comment]
mark

**Pull request merged**